### ISSUE TITLE
[Bug]: update_pnpm_lock fails under Bzlmod, retry fixes it

### ISSUE THREAD

**Issue by alexeagle** [aspect-build/rules_js#1445]
### What happened?

Under bzlmod, `update_pnpm_lock` fails on the first attempt.

### Version

HEAD

### How to reproduce

In rules_js,

1. `cd e2e/npm_translate_lock`
2. modify package.json, like `"semver": "7.5.4"`

then `bazel build ...` gives an error:

```shell
~/Projects/rules_js/e2e/npm_translate_lock$ bazel build --enable_bzlmod ...
Starting local Bazel server and connecting to it...
ERROR: Skipping '...': error loading package under directory '': error loading package '': at /shared/cache/bazel/user_base/3b36aba3ddc6cb653f7aaaed8bb07c4d/external/aspect_rules_js~override~npm~npm/defs.bzl:40:6: Unable to find package for @aspect_rules_js~override~npm~npm__semver__registry.npmjs.org_semver_7.5.4__links//:defs.bzl: The repository '@aspect_rules_js~override~npm~npm__semver__registry.npmjs.org_semver_7.5.4__links' could not be resolved: Repository '@aspect_rules_js~override~npm~npm__semver__registry.npmjs.org_semver_7.5.4__links' is not defined.
```

Simply running the same `bazel build` command again fixes it.

**Comment by alexeagle** [comment]
Adding some `print` debugging shows that the module extension isn't executed the way we thought. We call `npm_translate_lock' but it is non-blocking, the `module_ctx.read(attr.pnpm_lock)` later in `extensions.bzl` is already evaluated before we write the changed pnpm-lock.yaml back to the disk. I think this is fixable, but I'm not exactly sure how, and anyway I don't think we have time to prioritize it right now.

**Comment by jbedard** [comment]
Support for this has been removed (bzlmod and workspace) in 2.x: https://github.com/aspect-build/rules_js/commit/ca4a3cb74cf623ef0b5834f2847f093a0bb6f361

**Comment by nishant-satpathy** [comment]
has this bug been fixed? I am using the `2.4.2` version of this rule and still encounter this fail and retry behavior.

**Comment by jbedard** [comment]
See https://github.com/aspect-build/rules_js/issues/1445#issuecomment-2097167353
